### ISSUE TITLE
[Why are you just take last class]

### ISSUE THREAD

**Issue by Vietdung113** [LecJackS/CS-330-Deep-Multi-Task-and-Meta-Learning#4]
i have read the [code](https://github.com/LecJackS/CS-330-Deep-Multi-Task-and-Meta-Learning/blob/3b6162b302bd9cbff11d4b448c926a87f67a07f9/hw1/hw1.py#L34).
And i don't understand why are you just take last class for compute loss?
```   
loss = tf.keras.losses.categorical_crossentropy(y_true=labels[:, -1:, :, :], y_pred=preds[:, -1:, :, :],from_logits=True)
```
Can you explain it ?

**Comment by LecJackS** [comment]
From [Assignment 1 pdf](https://github.com/LecJackS/CS-330-Deep-Multi-Task-and-Meta-Learning/blob/master/hw1/homework_1_updated.pdf):

> # Problem 2: Memory Augmented Neural Networks [2, 3]
>
>We will be attempting few shot classification using memory augmented neural networks.
>
>The idea of memory augmented networks is to use a classifier with recurrent memory, such that information from the K examples of unseen classes informs classification through the hidden state of the network.
>
> The data processing will be done as in [SNAIL \[3\]](https://arxiv.org/abs/1707.03141). Specifically, during training, you sample batches of N classes, with K +1 samples per batch. Each set of labels and images are concatenated together, and then all K of these concatenated pairs are sequentially passed through the network.
>
>Then the final example of each class is fed through the network (concatenated with 0 instead of the true label). The loss is computed between these final outputs and the ground truth label, which is then backpropagated through the network.
>
>**Note: The loss is only computed on the last set of N classes.**
>
>The idea is that the network will learn how to **encode the first K examples of each class into memory such that it can be used to enable accurate classification on the K + 1th example**. See Figure 1.
>
> ![figure-1](https://i.imgur.com/v8Mt5ug.png)

So we are only interested in predicting the last class, not all the classes, we already have the class name in training, its passing as training data.

The network must learn some kind of function f st.

f ( [(x1,y1), (x2,y2), (x3,y3)] , x4) = ?

where ? should be one of those 3 clases: y1, y2, y3

That's why we only care about the loss over the last class of each meta batch.

Let me know if it's clear!

**Comment by Vietdung113** [comment]
thank you for your reply. 
I will close this issue
